soc/arm/nxp_imx/rt5xx/soc.c

```diff
@@ -512,6 +513,29 @@ void __weak imxrt_deinit_display_interface(void)
 
 #endif
 
+static int nxp_rt500_late_init(void)
+{
+#if DT_NODE_HAS_COMPAT_STATUS(DT_NODELABEL(isp), nxp_bootpin, okay)
+
+	int err;
+
+	/* Declare pin configuration state for isp pin here */
+	PINCTRL_DT_DEFINE(DT_NODELABEL(isp));
+
+	/* Apply pinctrl state directly, since there is no isp device driver */
+	err = pinctrl_apply_state(PINCTRL_DT_DEV_CONFIG_GET(DT_NODELABEL(isp)),
+				  PINCTRL_STATE_DEFAULT);
+
+	if (err) {
+		return err;
+	}
+
+#endif
+	return 0;
+}
+
+SYS_INIT(nxp_rt500_late_init, PRE_KERNEL_2, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
```

Comment:
please guard the whole function/SYS_INIT, there's no point in running an empty SYS_INIT.